packages/flutter_markdown/lib/src/syntax_highlighter.dart

```diff
@@ -0,0 +1,339 @@
+    'new', 'null', 'operator', 'part', 'rethrow', 'return', 'set', 'static',
+    'super', 'switch', 'sync', 'this', 'throw', 'true', 'try', 'typedef', 'var',
+    'void', 'while', 'with', 'yield'
+  ];
+
+  static const List<String> _kBuiltInTypes = const <String>[
+    'int', 'double', 'num', 'bool'
+  ];
+
+  String _src;
+  StringScanner _scanner;
+
+  List<_HighlightSpan> _spans;
+
+  TextSpan format(String src) {
+    _src = src;
+    _scanner = new StringScanner(_src);
+
+    if (_generateSpans()) {
+      // Successfully parsed the code
+      List<TextSpan> formattedText = <TextSpan>[];
+      int currentPosition = 0;
+
+      for (_HighlightSpan span in _spans) {
+        if (currentPosition != span.start)
+          formattedText.add(new TextSpan(text: _src.substring(currentPosition, span.start)));
+
+        formattedText.add(new TextSpan(style: span.textStyle(_style), text: span.textForSpan(_src)));
+
+        currentPosition = span.end;
+      }
+
+      if (currentPosition != _src.length)
+        formattedText.add(new TextSpan(text: _src.substring(currentPosition, _src.length)));
+
+      return new TextSpan(style: _style.baseStyle, children: formattedText);
+    } else {
+      // Parsing failed, return with only basic formatting
+      return new TextSpan(style:_style.baseStyle, text: src);
+    }
+  }
+
+  bool _generateSpans() {
+    int lastLoopPosition = _scanner.position;
+
+    while(!_scanner.isDone) {
+      // Skip White space
+      _scanner.scan(new RegExp(r"\s+"));
+
+      // Block comments
+      if (_scanner.scan(new RegExp(r"/\*(.|\n)*\*/"))) {
+        _spans.add(new _HighlightSpan(
+          _HighlightType.comment,
+          _scanner.lastMatch.start,
+          _scanner.lastMatch.end
+        ));
+        continue;
+      }
+
+      // Line comments
+      if (_scanner.scan("//")) {
+        int startComment = _scanner.lastMatch.start;
+
+        bool eof = false;
+        int endComment;
+        if (_scanner.scan(new RegExp(r".*\n"))) {
+          endComment = _scanner.lastMatch.end - 1;
+        } else {
+          eof = true;
+          endComment = _src.length;
+        }
+
+        _spans.add(new _HighlightSpan(
+          _HighlightType.comment,
+          startComment,
+          endComment
+        ));
+
+        if (eof)
+          break;
+
+        continue;
+      }
+
+      // Raw r"String"
+      if (_scanner.scan(new RegExp(r'r".*"'))) {
+        _spans.add(new _HighlightSpan(
+          _HighlightType.string,
+          _scanner.lastMatch.start,
+          _scanner.lastMatch.end
+        ));
+        continue;
+      }
+
+      // Raw r'String'
+      if (_scanner.scan(new RegExp(r"r'.*'"))) {
+        _spans.add(new _HighlightSpan(
+          _HighlightType.string,
+          _scanner.lastMatch.start,
+          _scanner.lastMatch.end
+        ));
+        continue;
+      }
+
+      // Multiline """String"""
+      if (_scanner.scan(new RegExp(r'"""(?:[^"\\]|\\(.|\n))*"""'))) {
```

Comment:
I don't think we should have these regular expressions.  I'd probably just remove the syntax highlighting feature.